src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,989 @@ const ActorInit Dm_Char08_InitVars = {
+        EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->bubblePos, &velocity, &sBubbelAccel, &sBubblePrimColor,
+                                          &sBubbleEnvColor, Rand_S16Offset(80, 130), 30, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFBA4.s")
+void func_80AAFCCC(DmChar08* this, GlobalContext* globalCtx) {
+    switch (globalCtx->sceneNum) {
+        case SCENE_31MISAKI:
+            if (!(gSaveContext.save.weekEventReg[0x37] & 0x80)) {
+                switch (this->unk_206) {
+                    case 0:
+                        return; // Must be a return
+                    case 1:
+                        Message_StartTextbox(globalCtx, 0x102A, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+                    case 2:
+                        if (Message_GetState(&globalCtx->msgCtx) == 2) {
+                            Message_StartTextbox(globalCtx, 0x102B, &this->dyna.actor);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            } else {
+                switch (this->unk_206) {
+                    case 0:
+                        break;
+                    case 1:
+                        Message_StartTextbox(globalCtx, 0x103B, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+                    case 2:
+                        if (Message_GetState(&globalCtx->msgCtx) == 2) {
+                            func_801477B4(globalCtx);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            }
+            break;
+        case SCENE_SEA:
+            switch (this->unk_206) {
+                case 0:
+                    break;
+                case 1:
+                    Message_StartTextbox(globalCtx, 0x102E, &this->dyna.actor);
+                    this->unk_206++;
+                    break;
+                case 2:
+                    if (Message_GetState(&globalCtx->msgCtx) == 2) {
+                        Message_StartTextbox(globalCtx, 0x102F, &this->dyna.actor);
+                        this->unk_206 = 0;
+                    }
+                    break;
+            }
+            break;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFCCC.s")
+void func_80AAFE78(DmChar08* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFE78.s")
+void func_80AAFE88(DmChar08* this, GlobalContext* globalCtx) {
+    s32 actorActionIndex;
+    CsCmdActorAction* csAction;
+    f32 phi_f12;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFE88.s")
+    if (Cutscene_CheckActorAction(globalCtx, 0x1DAU)) {
+        actorActionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x1DA);
```

Comment:
We are using dec for the action cmd
```suggestion
    if (Cutscene_CheckActorAction(globalCtx, 474)) {
        actorActionIndex = Cutscene_GetActorActionIndex(globalCtx, 474);
```